### ISSUE TITLE
Make `Select.ready()` an async iterable

### ISSUE THREAD

**Issue by leandro-lucarella-frequenz** [frequenz-floss/frequenz-channels-python#47]
If we do this, the usage can be much more idiomatic:

```py
    receiver1 = some_chan.get_receiver()
    timer = channel.Timer(30.0)
    select = Select(receiver1=receiver1, timer=timer)
    async for selected in select.ready():
        match selected.origin:
            case receiver1:
                process_data(selected.value)
            case timer:
                # something to do once every 30 seconds
                pass
            case _:
                raise Exception("Unknown origin")
```

So `.ready()` should return an async iterator itself, by just `yield`ing for each ready message.

With this we also don't need to keep a reference to the `Select` instance around, we can just do `async for selected in Select(...).ready()`, or we could just add an utility function to make it even more compact, as 99% of the cases nobody will need to even know `Select` is a class: `async for selected in select(...)`. This also completely removes the temptation of adding the option to dynamically change the `Select` instance inside the loop (like in #24).

This also sort of depends on #27, as it could make breaking from the loop even more natural, and cause data loss in that case.

_Originally posted by @leandro-lucarella-frequenz in https://github.com/frequenz-floss/frequenz-channels-python/issues/27#issuecomment-1303299010_

**Comment by leandro-lucarella-frequenz** [comment]
This will be probably dropped because of the issues mentioned in:

* https://github.com/frequenz-floss/frequenz-channels-python/issues/27#issuecomment-1319252644.

**Comment by shsms** [comment]
I was just reading about the `Select` implementation in go again,  and realized that when there are multiple receivers ready,  it picks one at random,  and doesn't allow us to prioritize.

Now that I've thought about it further,  I think that's how it should be,  and is the true async way.  So maybe we should revisit the ideas in #27.

I feel it might be useful to look into what other languages do as well.  

Does that make sense?

**Comment by leandro-lucarella-frequenz** [comment]
I am not sure why would we want to not allow the user to prioritize what to process first if there are multiple receivers ready. :thinking: 

If it is some sort of safety (prevent starvation), maybe that should be an option?

```py
async for receiver in select.one_by_one():
    if receiver is x:
        pass
```

In any case (I mean if we went with a *one receiver at the time*) I would still prefer to offer an async iterator interface rather than the less pythonic `while` loop.

We could decide which one should be default, but removing options from the user doesn't seem like a good idea to me.

About looking at other languages, sure!

**Comment by leandro-lucarella-frequenz** [comment]
OK, now that I went though your new comments in the PR I have a better feeling for the starving issue and I think I'm starting to lean towards reviving #27 too. I created a new issue about that:

* #72
